types/src/transaction/mod.rs

```diff
@@ -592,16 +592,52 @@ impl From<VMStatus> for TransactionStatus {
     }
 }
 
+#[derive(Clone, Debug, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub enum GovernanceRole {
+    LibraRoot,
+    TreasuryCompliance,
+    Validator,
+    ValidatorOperator,
+    DesignatedDealer,
+}
+
+impl GovernanceRole {
+    pub fn from_role_id(role_id: u64) -> Option<Self> {
+        use GovernanceRole::*;
+        match role_id {
+            0 => Some(LibraRoot),
+            1 => Some(TreasuryCompliance),
+            2 => Some(DesignatedDealer),
+            3 => Some(Validator),
+            4 => Some(ValidatorOperator),
+            _ => None,
+        }
+    }
+
+    pub fn priority(&self) -> u64 {
+        use GovernanceRole::*;
+        match self {
+            LibraRoot => 2,
+            TreasuryCompliance => 1,
+            Validator | ValidatorOperator | DesignatedDealer => 0,
+        }
+    }
+}
+
 /// The result of running the transaction through the VM validator.
 #[derive(Clone, Debug, Eq, PartialEq)]
 pub struct VMValidatorResult {
     status: Option<DiscardedVMStatus>,
     score: u64,
-    is_governance_txn: bool,
+    governance_role: Option<GovernanceRole>,
```

Comment:
Thoughts on: 
- Having the `GovernanceRole` enum include all the roles (or one catch-all `NonGovernance` role)
- Getting rid of the `Option` here? 

I think that would let us simplify the comparison of `priority` in the mempool code and make it easier to test that `None` < `Some(GovernanceRole)`.

Reply:
I like this idea a lot more than the option. I'll update and add a `NonGovernanceRole` variant.